### PR TITLE
Python3 compatibility -- almost there (replace for #47)

### PR DESCRIPTION
**Pull request by yarikoptic**
Sits on top of https://github.com/moloney/dcmstack/pull/47 but also enables travis testing for python 3.5.
Seems that only one issue is still flaky with loading some array fields, didnt look in detail. Otherwise tests seems to pass

### CODE REVIEW

File: src/dcmstack/extract.py [hunk 1]
```diff
@@ -23,15 +25,22 @@
...
 def is_ascii(in_str):
     '''Return true if the given string is valid ASCII.'''
-    if all(' ' <= c <= '~' for c in in_str):
+    # We do not want to deal with encode/decode ATM, so let's assume
+    # that no fancy unicode is in there and just do blunt comparison char
+    # by char
+    start, end = ' ', '~'
+    if not PY2 and isinstance(in_str, bytes):
+        start, end = 32, 126
+    if all(start <= c <= end for c in in_str):
```
Comment (yarikoptic) [inline src/dcmstack/extract.py]: this is all a questionable approach/shortcut I took here ;-)

**Comment by yarikoptic** [comment]
as promised -- only few tests fail now under 3.5: https://travis-ci.org/moloney/dcmstack/jobs/402939084#L605

**Comment by yarikoptic** [comment]
ok -- part of the problem is not within dcmstack but within nibabel :-(  https://github.com/nipy/nibabel/pull/647

**Comment by yarikoptic** [comment]
whoohoo -- and here is with coverage+codecov reporting!  might need to enable integration at codecov.io to be in full effect, but overall all green so far with the workaround for nibabel.  please review/accept @bmoloney

**Comment by moloney** [comment]
Can you take a look at my branch "py3-compat" and merge here if everything looks good?  It improves test coverage and fixes some more issues.

**Comment by yarikoptic** [comment]
Thanks, will do!
Note that you could pay as well into this branch of mine (in my clone) since I allowed edits from maintainers

**Comment by yarikoptic** [comment]
pushed, along with just a tiny fix in bbbf3fbeffba222bdf4560316ba7383eb69d3890 
didn't have yet a chance for a thorough pass, but let's see first how tests behave I guess

**Comment by KirstieJane** [comment]
Hey @moloney & @yarikoptic! This is a super useful PR so I'm just swinging by to say it would be really awesome to have merged! @islast in my lab has been working from it today with [heudiconv](https://github.com/nipy/heudiconv) so if you need any help with testing etc we could maybe give some feedback? I think its working well so far but it would be really cool to be able to pip install rather than pull from github.

I totally understand you're both really busy so please disregard this ping until you have time to look at it! Thank you!!

**Comment by yarikoptic** [comment]
Woohoo!!!! Thanks!
Now we will just need a release ;-)

**Pull request merged**